### DISCUSSION TITLE
`spool_max_size` is ignored in `UploadFile`

### DISCUSSION THREAD

**Discussion by agronick** [fastapi/fastapi#6044] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import fastapi


@router.post("/upload_blog_articles")
async def injest_blog(
    file: fastapi.UploadFile = fastapi.File(..., media_type="text/csv"),
):
```

### Description

In the init method of `UploadFile` in starlettte `file` is always `None`. That means no matter how big the file is we are getting a spooled file instead of a regular file.

```py
    def __init__(
        self,
        filename: str,
        file: typing.Optional[typing.BinaryIO] = None,
        content_type: str = "",
        *,
        headers: "typing.Optional[Headers]" = None,
    ) -> None:
        self.filename = filename
        self.content_type = content_type
        if file is None:
            self.file = tempfile.SpooledTemporaryFile(max_size=self.spool_max_size)  # type: ignore  # noqa: E501
        else:
            self.file = file
        self.headers = headers or Headers()
```

### Operating System

macOS

### FastAPI Version

0.87.0

### Python Version

3.10.4

**Comment by jgould22** [comment, marked as answer]
Only until it reaches 1mb, then it is written to disk and becomes a normal TemporaryFile

https://github.com/encode/starlette/blob/master/starlette/datastructures.py#L431

**Comment by agronick** [comment]
I subclassed it and set it to 0 and never got a real file.

**Comment by jgould22** [comment]
Can you expand on what you mean by "never got a real file" ?

wait, if you set it to 0 the Spooled File will never roll over and it will never write the file to disk

**Comment by agronick** [comment]
I couldn't open it or chose the mode. I had a bad csv file that only worked with `csv.reader` when it was given a file like object. In a thread I had to make a`NamedTemporaryFile` with mode `r+`, write the bytes to that decoded to utf-8, seek to 0, and give that object to the csv reader. It didn't like `file` or `file.file` (when the type of was set to `UploadFile`) or `bytes.decode("utf-8")` (when the type was set to `bytes`). That was the only way I could make it work.
